### ISSUE TITLE
make opening .ipynb files instaneous

### ISSUE THREAD

**Issue by danielkrajnik** [millejoh/emacs-ipython-notebook#873]
Opening .ipynb files should get you straight away to the rendered notebook. No steps in-between. No manual server starting. No buffer switching. Just open file = open notebook.

There is no point  for the user to start server and switch buffer - vscode jupyter extension gets you directly to the notebook... why would you make it so complicated to the users?

It's such a bad idea, please can we change it?

**Comment by dickmao** [comment]
> It's such a bad idea, please can we change it?

tl;dr
```
(add-hook 'ein:ipynb-mode-hook
          (lambda ()
            (cl-letf (((symbol-function 'read-directory-name)
                       (lambda (_prompt dir &rest _args) dir)))
              (ein:process-find-file-callback))))
```

29 out of 30 people would agree with you, which is why roughly that ratio of the computing public uses vscode.

The exception is someone with a strong UNIX background who would find your presumption about skipping steps distasteful.

That person would also find distasteful the entire premise of a json format impervious to version control that also requires rendering by an additional moving part of a server or viewer.

This is not an endorsement of org-babel which I also find obtuse.  For whatever reason, it's not cool anymore to just write a python script.

In my defence, I first encountered jupyter notebooks during graduate school before I knew what git was.

**Comment by danielkrajnik** [comment]
perfect, thank you very much! it would have taken me ages to figure out how to do it.

It doesn't quite work - could you please help me focus on this buffer?
![image](https://github.com/millejoh/emacs-ipython-notebook/assets/15096799/e6a8a49a-44b5-4bae-a358-04234886ec04)

**Comment by danielkrajnik** [comment]
If someone wants to understand what the snippet is doing these may be helpful:
https://stackoverflow.com/questions/869529/difference-between-set-setq-and-setf-in-common-lisp

**Comment by danielkrajnik** [comment]
**and most importantly all of this** (because ein devs are too stupid to give you basic functionality so you have to do it yourself):
**https://github.com/chrisdone/elisp-guide**

**Comment by danielkrajnik** [comment]
I can't believe that you used `cl-letf`  for your example. This is a total dick move. You will do anything to overcomplicate it. That's why nobody is using emacs.

**Comment by danielkrajnik** [comment]
Fuck it, I give up. ein is a mess
